core/src/main/java/org/apache/accumulo/core/crypto/CryptoServiceFactory.java

```diff
@@ -25,11 +26,38 @@
 
   /**
    * Create a new crypto service configured in {@link Property#INSTANCE_CRYPTO_SERVICE}.
+   *
+   * @param useAccumuloClassloader
+   *          Determine if Accumulo class loader is used. When calling from accumulo server code,
+   *          pass true. When calling from Accumulo client code, pass false.
    */
-  public static CryptoService newInstance(AccumuloConfiguration conf) {
-    CryptoService newCryptoService = Property.createInstanceFromPropertyName(conf,
-        Property.INSTANCE_CRYPTO_SERVICE, CryptoService.class, new NoCryptoService());
+  public static CryptoService newInstance(AccumuloConfiguration conf,
+      boolean useAccumuloClassloader) {
+
+    CryptoService newCryptoService;
+
+    if (useAccumuloClassloader) {
+      newCryptoService = Property.createInstanceFromPropertyName(conf,
+          Property.INSTANCE_CRYPTO_SERVICE, CryptoService.class, new NoCryptoService());
+    } else {
+      String clazzName = conf.get(Property.INSTANCE_CRYPTO_SERVICE);
+      if (clazzName == null || clazzName.trim().isEmpty()) {
+        newCryptoService = new NoCryptoService();
+      } else {
+        try {
+          newCryptoService = CryptoServiceFactory.class.getClassLoader().loadClass(clazzName)
+              .asSubclass(CryptoService.class).newInstance();
+        } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+          throw new RuntimeException(e);
+        }
+      }
+    }
+
     newCryptoService.init(conf.getAllPropertiesWithPrefix(Property.INSTANCE_CRYPTO_PREFIX));
     return newCryptoService;
   }
+
+  public static CryptoService newDefaultInstance() {
+    return newInstance(DefaultConfiguration.getInstance(), false);
```

Comment:
I think this should get what is configured and not just the default.  Calling newDefaultInstance is the same as using NoCryptoService, which is what we have now but may break special parts of the code when crypto is configured.

Reply:
> I think this should get what is configured and not just the default.

Configured where?  The purpose of this method is for code that has no meaningful config can call it.

>  Calling newDefaultInstance is the same as using NoCryptoService

At first I was creating NoCryptoService everywhere in the code.  Then I realized that if we ever changed the default that all of this code would need to be changed.  So I created this method which at first just returned `new NoCryptoService()`.  Then I realized that if we ever change the default that we will need to change this method also.  So I changed it to use the default config.  That way if the default is ever changed this code does not need to change.

Reply:
I was just concerned that DefaultConfiguration.getInstance() is calling static code that we are trying to avoid.  But if its not, then yeah it would be better to get the default everywhere.